### ISSUE TITLE
Pawns not promoted properly in algebraicGameClient.js

### ISSUE THREAD

**Issue by CFT-Chris** [brozeph/node-chess#18]
I encountered this bug a while back so hopefully my thinking is still accurate.

Steps to reproduce: underpromote any pawn or explicitly promote the pawn to a queen (i.e. append "Q" to notation)

Issue: Pawn is not promoted at all.

Suggested Fix:
```javascript
		// check for pawn promotion
		if (notation.charAt(notation.length - 1).match(/[BNQR]/)) {
			promo = notation.charAt(notation.length - 1);
            notation = notation.slice(0, notation.length - 1); // <------- remove it from the notation
		}
```

Basically, if I can recall correctly, the notation when not stripped of the promotion character will cause the function to recurse on line 267 and return immediately, and thus never hit the promotion code on lines 275 and onward.  Taking away the promo character allows it to fall through to the promotion code.

**Comment by CFT-Chris** [comment]
My memory is a lot better after looking at that other issue I listed.  This one is actually related to that issue #17 

Here's the test case, similar to #17 to reproduce it:

```javascript
var chess = require('chess');
var gc = chess.create();
var piece = require('chess/lib/piece');
var status;

  // position the board for a promotion next move
  gc.game.board.getSquare('c7').piece = null;
  gc.game.board.getSquare('c2').piece = null;
  gc.game.board.getSquare('c7').piece = piece.createPawn(piece.SideType.White);
  gc.game.board.getSquare('c7').piece.moveCount = 1;

  status = gc.getStatus(true);

  console.log(Object.keys(status.notatedMoves));
```

which outputs (note the missing promotions on the pawn):

```
[ 'Na3',
  'Nc3',
  'Qc2',
  'Qb3',
  'Qa4',
  'Nf3',
  'Nh3',
  'a3',
  'a4',
  'b3',
  'b4',
  'd3',
  'd4',
  'e3',
  'e4',
  'f3',
  'f4',
  'g3',
  'g4',
  'h3',
  'h4',
  'cxb8',
  'cxd8' ]
```

My suggested fix from the first post is NOT the correct fix.  A better fix would be in notate of algebraicGameClient.js line 91:
```javascript
// squares with pawns
			if (p.type === piece.PieceType.Pawn) {
        if (sq.piece)
          prefix = validMoves[i].src.file;
        
        // check for promotion
        if (p.type === piece.PieceType.Pawn) {
          isPromotion = (sq.rank === 8 || sq.rank === 1);
			// squares with Bishop, Knight, Queen or Rook pieces
			} else if (p.type === piece.PieceType.Bishop ||
```

or something similar, where it is possible to notate the capture and the promotion at once for a pawn.

**Comment by brozeph** [comment]
@CFT-Chris - Another great find, thank you! This is now fixed and published to NPM in `v0.2.7` of the module.
